src/mlpack/methods/ann/layer/sigmoid_cross_entropy_error_impl.hpp

```diff
@@ -0,0 +1,84 @@
+/**
+ * @file sigmoid_cross_entropy_error_impl.hpp
+ * @author Kris Singh
+ *
+ * Implementation of the sigmoid cross entropy error performance function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_SIGMOID_CROSS_ENTROPY_ERROR
+#define MLPACK_METHODS_ANN_LAYER_SIGMOID_CROSS_ENTROPY_ERROR
+
+// In case it hasn't yet been included.
+#include "sigmoid_cross_entropy_error.hpp"
+#include <mlpack/methods/ann/activation_functions/softplus_function.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename InputDataType, typename OutputDataType>
+SigmoidCrossEntropyError<InputDataType, OutputDataType>
+::SigmoidCrossEntropyError()
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename eT>
+inline double SigmoidCrossEntropyError<InputDataType, OutputDataType>::Forward(
+    const arma::Mat<eT>&& input, const arma::Mat<eT>&& target)
+{
+  double max = 0;
+  arma::mat output;
+  SoftplusFunction::Fn(static_cast<arma::mat>(-arma::abs(input)), output);
```

Comment:
I realize this was not your code originally, but I think this can be removed.  Keeping in mind that all we want in the end is `arma::accu(output)`, and `output` is made by looping over the elements of `input` individually, we can simply add the elementwise calculation to the loop below:

```
max += SoftplusFunction::Fn(-std::abs(input[i]));
```

and maybe it would be worthwhile to change the name of the `max` variable.

(Another note: the static_cast<> is a bad idea and could cause some really strange problems.  The code I'm suggesting above is limited to `eT = double`, but we can leave that for #1062.)